### PR TITLE
fix(api): validate test run spec name lengths to handle overflow

### LINKED ISSUE #173: Create test run API returns 500 for spec name length > 255
The Create Test Run API responds with a 500 Internal Server Error when any spec name exceeds 255 characters.

This is due to database constraint (varchar(255)). 

```
[FAILED] Unable to push report to Fern: client: Response status code: 500, Response: {"error":"failed to create test run: failed to create test run: ERROR: value too long for type character varying(255) (SQLSTATE 22001)"}
  Expected
      <*errors.errorString | 0x1400028c9f0>:
      client: Response status code: 500, Response: {"error":"failed to create test run: failed to create test run: ERROR: value too long for type character varying(255) (SQLSTATE 22001)"}
      {
          s: "client: Response status code: 500, Response: {\"error\":\"failed to create test run: failed to create test run: ERROR: value too long for type character varying(255) (SQLSTATE 22001)\"}",
      }
  to be nil
```

The API should return a 400 Bad Request with a validation error indicating that spec names must not exceed 255 characters.

### PR DESCRIPTION
**Pull request by Naganathan05**
Fixes #173 
Added validation in create test run api to ensure spec names do not exceed 255 characters before DB insertion. Now returns `400` Bad Request instead of `500` Internal Server Error for invalid input which was due to constraint `varchar(255)`

---
## Summary by cubic
Validate test run spec names (max 255 runes) and return 400 for invalid input across all create paths to prevent DB overflow and clarify client errors.

- **Bug Fixes**
  - Add `ValidateTestRun` with rune-based checks; ignore nil specs; reject nil `testRun`.
  - Pre-validate in `CreateTestRun` and `CreateTestRunWithSuites` before any DB writes.
  - Map `domain.ErrInvalidTestRun` to HTTP 400 in admin create, public record, and adapter `CreateTestRunWithSuites`; add tests for unicode overlength, nil cases, and 400 responses.

<sup>Written for commit 16c334e1d3406ecfede5696e118a237f8b1569be. Summary will update on new commits.</sup>

### CODE REVIEW

File: internal/api/test_run_handler.go [hunk 1]
```diff
@@ -87,6 +88,10 @@ func (h *TestRunHandler) createTestRun(c *gin.Context) {
 
 	// Create test run using domain service
 	if _, _, err := h.testingService.CreateTestRun(c.Request.Context(), testRun); err != nil {
+		if strings.Contains(err.Error(), "invalid test run") {
```
Comment (bsekar) [inline internal/api/test_run_handler.go]: There are 3 other codepaths that also throw "invalid test run" , If any of those errors ever bubble up through one of the three handler call sites (likely as the codebase evolves), they'll be silently misclassified as 400 instead of 500.

**Review by bsekar — DISMISSED** [review]

**Review by bsekar — APPROVED** [review]
